packages/fluentui/react-northstar/src/components/Datepicker/DatepickerCalendar.tsx

```diff
@@ -197,31 +256,53 @@ export const DatepickerCalendar: ComponentWithAs<'div', DatepickerCalendarProps>
             ...unhandledProps,
           })}
         >
+          {renderHeader()}
           <Grid rows={grid.length + 1} columns={DAYS_IN_WEEK}>
-            {_.times(DAYS_IN_WEEK, dayNumber => (
-              <Text
-                key={`header ${dayNumber}`}
-                align="center"
-                content={localizedStrings.shortDays[(dayNumber + firstDayOfWeek) % DAYS_IN_WEEK]}
-              />
-            ))}
-            {_.map(grid, week =>
-              _.map(week, (day: IDay) => {
-                return (
-                  <Button
-                    key={day.key}
-                    content={day.date}
-                    aria-label={`${formatMonthDayYear(day.originalDate, localizedStrings)}`}
-                    onClick={e => {
-                      _.invoke(props, 'onDateChange', e, { ...props, value: day });
-                    }}
-                    primary={day.isSelected}
-                    disabled={!day.isInMonth}
-                    text
-                  />
-                );
+            {_.times(DAYS_IN_WEEK, dayNumber =>
+              createShorthand(DatepickerCalendarHeaderCell, calendarHeaderCell, {
+                defaultProps: () =>
+                  getA11yProps('calendarHeaderCell', {
+                    className: datepickerCalendarHeaderCellClassName,
+                    content: localizedStrings.shortDays[(dayNumber + firstDayOfWeek) % DAYS_IN_WEEK],
+                    key: `header_${dayNumber}`,
+                  }),
               }),
             )}
+            {_.map(grid, week =>
+              _.map(
+                week,
+                (day: IDay) =>
+                  createShorthand(DatepickerCalendarCell, calendarCell, {
+                    defaultProps: () =>
+                      getA11yProps('calendarCell', {
+                        className: datepickerCalendarCellClassName,
+                        content: day.date,
+                        key: day.key,
+                        'aria-label': formatMonthDayYear(day.originalDate, localizedStrings),
```

Comment:
Shouldn't this be in the behavior?

Reply:
Maybe, I'm still thinking on that one. And I'm planning to work on the behaviors in the next PR, so the reviews are split :)